### ISSUE TITLE
Question about Bias & Weight initialization

### ISSUE THREAD

**Issue by LeoCho1learning** [ultralytics/yolov5#2429]
## ❔Question
In detect module, biases are initialized following sec. 3.3 of focal loss, but in a diferent way. So I want to ask how exactly these params are choosed like (8 / (640 / s) ** 2). It seems that 8 is obtained by statistics saying that there would have 8 objects in a 640x640 image. But why?

And in focal loss, weights are setted under nomal distribution. But in yolov5, weights are setted in default way which is kaiming_uniform instead of kaiming_normal. I can't figure this out either.

Thanks.

**Comment by glenn-jocher** [comment]
@LeoCho1learning 8 instances per image is the COCO mean.

We don't employ focal loss here by default, though you can enable it in the hyperparameter yaml file.

**Comment by LeoCho1learning** [comment]
> @LeoCho1learning 8 instances per image is the COCO mean.
> 
> We don't employ focal loss here by default, though you can enable it in the hyperparameter yaml file.

Thanks for your reply. 
So the initialization of weight should be kaiming_normal if I try to use focal loss. Am i right?
And 8 is the mean numbers of instances per image in coco. And (640 / s) ** 2 means the size of feature map of each output stage. I think v5 want to give biases of detect a small value to overcome the inbalance issue at early time of training. But i don't understand the reason of mmaking it equal to the ratio between mean numbers of instances and featuremap size.

Thanks

**Comment by glenn-jocher** [comment]
@LeoCho1learning well as you said if you use focal loss you may want to adjust the initialization strategy as well, assuming you train from scratch, otherwise the strategy will matter little to nothing.

We are setting the objectness neurons to their projected mean values at training start. To do this you need to relate the output anchor count to the instance count, and take into account the average number of anchors that match to a label, etc.

**Comment by LuthraBhomik** [comment]
@glenn-jocher  If I am training the images using size 320x320 instead of 640x640, do I need to change here : 

b.data[:, 4] += math.log(8 / (640 / s) ** 2)

Can you please explain whats happening here ?

**Comment by glenn-jocher** [comment]
@LuthraBhomik bias init, very minor effect on early epochs. You can modify if you want.
